designer/src/components/field-editor.tsx

```diff
@@ -149,6 +178,48 @@ export const FieldEditor = ({
     }
   };
 
+  /**
+   * Finds where a field is used in conditions or templated string fields
+   */
+  const findFieldUsage = (
+    fieldName: string,
+    allFields: Record<string, any>,
```

Comment:
Cool, I've not come across this before, very useful. Shame about the name though since we have many Record* types!